src/main/java/org/apache/sling/testing/clients/query/servlet/QueryServlet.java

```
@@ -0,0 +1,167 @@
+            boolean showResults = !("false".equalsIgnoreCase(request.getParameter("showresults")));
+            // default for explainQuery is false, unless parameter is present and is not matching "false"
+            String explainParam = request.getParameter("explain");
+            boolean explainQuery = (explainParam != null) && !("false".equalsIgnoreCase(explainParam));
+
+            boolean tidy = false;
+            for (String selector : request.getRequestPathInfo().getSelectors()) {
+                if ("tidy".equals(selector)) {
+                    tidy = true;
+                }
+            }
+
+            if ((query == null) || query.equals("") || (type == null) || type.equals("")) {
+                response.sendError(400, "Parameters query and type are required"); // invalid request
+                return;
+            }
+
+            // prepare
+            if (explainQuery) {
+                query = "explain " + query;
+            }
+            javax.jcr.query.Query q = qm.createQuery(query, type);
+
+            // execute
+            before = new Date().getTime();
+            QueryResult result = q.execute();
+            after = new Date().getTime();
+
+            // collect results
+            String firstSelector = null;
+            if (result.getSelectorNames().length > 1) {
+                firstSelector = result.getSelectorNames()[0];
+                try {
+                    String[] columnNames = result.getColumnNames();
+                    if (columnNames.length > 0) {
+                        String firstColumnName = columnNames[0];
+                        int firstDot = firstColumnName.indexOf('.');
+                        if (firstDot > 0) {
+                            firstSelector = firstColumnName.substring(0, firstDot);
+                        }
+                    }
+                } catch (Exception e) {
+                    //log.warn("Could not get selector name of the first column", e);
```

Why is this commented out? Don't we need to know about it?

---

It's an expected case, handled later by checking null. I've removed the comment.

---

Well, what could happen to trigger the exception? I'd rather check for it the probable causes and avoid them rather than have an empty catch block.